include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -0,0 +1,415 @@
+        data = sdsl::bit_vector(bits_);
+        bin_width   = (bins + 63) >> 6;
+        block_size  = bin_width << 6;
+        block_count = bits_ / block_size;
+
+        pre_hash.resize(num_hash);
+
+        for (size_t i = 0; i < num_hash; ++i)
+            pre_hash[i] = i ^ (15 * seed);
+    }
+
+    /*!\brief Construct a compressed Interleaved Bloom Filter.
+     * \param[in] ibf The uncompressed seqan3::interleaved_bloom_filter.
+     *
+     * \attention This constructor can only be used to construct **compressed** Interleaved Bloom Filters.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_constructor_compressed.cpp
+     */
+    interleaved_bloom_filter(interleaved_bloom_filter<data_layout::uncompressed> const & ibf)
+    //!\cond
+        requires data_layout_mode == data_layout::compressed
+    //!\endcond
+    {
+        std::tie(bins, bin_width, block_size, block_count, num_hash, pre_hash) =
+            std::tie(ibf.bins, ibf.bin_width, ibf.block_size, ibf.block_count, ibf.num_hash, ibf.pre_hash);
+
+        data = sdsl::sd_vector<>{ibf.data};
+    }
+    //!\}
+
+    /*!\name Data access
+     * \{
+     */
+    /*!\brief Inserts a value into a specific bin.
+     * \param[in] h   The raw hash value to process.
+     * \param[in] bin The bin to insert into.
+     *
+     * \attention This function is only available for **uncompressed** Interleaved Bloom Filters.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_set.cpp
+     */
+    inline void set(size_t const h, size_t const bin)
+    //!\cond
+        requires data_layout_mode == data_layout::uncompressed
+    //!\endcond
+    {
+        assert(bin < bins);
+        for (auto const & val : pre_hash)
+        {
+            size_t idx = val * h;
+            hash_and_fit(idx);
+            idx += bin;
+            assert (idx < data.size());
+            data[idx] = 1;
+        };
+    }
+
+    /*!\brief Determines set membership of a given value.
+     * \param[in,out] result The sdsl::bit_vector to store the result into.
+     * \param[in]     h      The raw hash value to process.
+     *
+     * \attention The size `result` must be equal to `get_bin_number()`. In debug mode an assertion checks this.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_get.cpp
+     */
+    void get(size_t const h, sdsl::bit_vector & result) const
+    {
+        assert(result.size() == get_bin_number());
+
+        std::vector<size_t> idx = pre_hash;
+
+        std::ranges::for_each(idx, [this, h] (size_t & i) { i *= h; hash_and_fit(i); });
+
+        for (size_t batch = 0; batch < bin_width; ++batch)
+        {
+           assert(idx[0] < data.size());
+           size_t tmp = data.get_int(idx[0]);
+           idx[0] += 64;
```

Comment:
Why add 64?

Reply:
Because we read the first 64 bins, so in the next iteration, if there is one, we need to read the next 64 bins